### PR TITLE
Fixed two critical bugs in get_joint_graph

### PR DESCRIPTION
**Pull request by kaozkai**
1.) When clustering two repertoires with two chains, CDR3b edges between-repertoire got deleted due to typo in helper function get_ix() 

2.) Duplicated edges removal with which_multiple removed also the CDR3b edge of double edges between-repertoire clones

Also, added test-multi-chain-clustering to testthat

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed CDR3 edge clustering to properly identify identical sequences in all scenarios
  * Prevented unintended deletion of CDR3b edges between repertoires
  * Improved handling of double-edge cases in multi-chain clone comparison

* **Tests**
  * Added comprehensive test suite for multi-chain clustering validation

### CODE REVIEW

File: R/graph.R [hunk 1]
```diff
@@ -221,15 +220,17 @@ get_joint_graph <- function(clust_irrs, cores = 1) {
...
-        g <- igraph::delete_edges(graph = g, edges = which(de))
+    # delete duplicated edges, excluding those between different chains
+    if("chain" %in% names(df_e)) {
+        df_e <- df_e[!duplicated(df_e[, c("from", "to", "chain")]), ]
+    } else {
+        df_e <- df_e[!duplicated(df_e[, c("from", "to")]), ]
     }
     
+    # build joint graph
+    g <- graph_from_data_frame(df_e, directed = FALSE, vertices = df_v)
```
[the remark this answers is not in the document]
Comment (kaozkai) [reply R/graph.R]: Ok... I will check if we do in fact have at any point directed graphs (that would be turned into undirected graphs in the very next line), and adapt the deduplication if necessary.

**Comment by snaketron** [comment]
> 2.) Duplicated edges removal with which_multiple removed also the CDR3b edge of double edges between-repertoire clones

I don't get this point. Please explain in more detail. At the end of the graph joining we want  to have a single edge (per chain) between two nodes. No?

**Comment by snaketron** [comment]
> 1.) When clustering two repertoires with two chains, CDR3b edges between-repertoire got deleted due to typo in helper function get_ix()

This is a good point!

**Comment by kaozkai** [comment]
> > 2.) Duplicated edges removal with which_multiple removed also the CDR3b edge of double edges between-repertoire clones
> 
> I don't get this point. Please explain in more detail. At the end of the graph joining we want to have a single edge (per chain) between two nodes. No?

Yes, one edge per connected chain. What I noticed (and wanted to say) is: 
if we have two edges between two identical clones between-repertoires, 
the _which_multiple()_ function does remove one of the two edges, because it only looks for the names and does not take the chain column into consideration.

I noticed this because the tests I wrote still failed for this case after the 1. point was corrected (see _tests/testthat/test-multi-chain-clustering.R_)
